### PR TITLE
Tab Top Rated

### PR DESCRIPTION
**Pull request by julian-mainardis-globant**
## Description

Added the Api service and Database structure. In this PR for the tab "Top Rated"

## Story Type

- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?

Playing the emulator and changing the tabs

## Evidences

| Pixel 3  | Nexus S |
| ------------- | ------------- |
| pixel3 | nexusS |

## Coverage
image
image
image

## Checklist

- All the unit test passed :white_check_mark:

## Tested On
- Pixel 3
- Nexus S

## Package Structure
image image

### CODE REVIEW

**Review by mariano98montero — APPROVED** [review]

File: data/src/main/java/com/example/data/database/MovieDao.kt [hunk 1]
```diff
@@ -5,12 +5,19 @@ import androidx.room.Insert
...
-    @Query("SELECT * FROM movie_table")
-    fun getNowPlayingMovies(): List<MovieEntity>
+    @Query("SELECT * FROM movie_table WHERE tab = :tabsEnum")
+    fun getNowPlayingMovies(tabsEnum: TabsEnum): List<MovieEntity>
 
     @Insert(onConflict = OnConflictStrategy.REPLACE)
     fun insertNowPlayingMovie(movieEntity: MovieEntity)
+
+    @Query("SELECT * FROM movie_table WHERE tab = :tabsEnum")
+    fun getTopRatedMovies(tabsEnum: TabsEnum): List<MovieEntity>
```
Comment (SartiMau) [inline data/src/main/java/com/example/data/database/MovieDao.kt]: this method is not necessary as you are doing the same in the lines 12 method

File: data/src/main/java/com/example/data/database/MovieDao.kt [hunk 2]
```diff
@@ -5,12 +5,19 @@ import androidx.room.Insert
...
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertTopRatedMovie(movieEntity: MovieEntity)
```
Comment (SartiMau) [inline data/src/main/java/com/example/data/database/MovieDao.kt]: this method is not necessary as you are doing the same in the lines 18 method

File: data/src/main/java/com/example/data/database/MovieRepositoryImpl.kt [hunk 3]
```diff
@@ -23,4 +24,19 @@ class MovieRepositoryImpl(private val movieDao: MovieDao) : MovieRepository {
             movieDao.insertNowPlayingMovie(it.transformToMovieEntity())
         }
     }
+
+    override fun getTopRatedMovies(): Result<List<Movie>> =
```
Comment (SartiMau) [inline data/src/main/java/com/example/data/database/MovieRepositoryImpl.kt]: these 2 methods are the same that the methods above. You can abstract the functionality by passing the movie type by param and reusing the same methods

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/example/data/database/MovieDao.kt (hunk 1 above)
Comment (julian-mainardis-globant) [reply data/src/main/java/com/example/data/database/MovieDao.kt]: it's true. fixed

File: data/src/main/java/com/example/data/database/MovieDao.kt (hunk 2 above)
Comment (julian-mainardis-globant) [reply data/src/main/java/com/example/data/database/MovieDao.kt]: it's true. fixed

File: data/src/main/java/com/example/data/database/MovieRepositoryImpl.kt (hunk 3 above)
Comment (julian-mainardis-globant) [reply data/src/main/java/com/example/data/database/MovieRepositoryImpl.kt]: it's true. fixed

**Review by SartiMau — APPROVED** [review]

**Review by Georgina-Fernandez — APPROVED** [review]

**Pull request merged**